Sources/Crust/Parser.swift

```diff
@@ -119,10 +124,16 @@ public class Parser {
 extension Parser {
   public func parseTopLevelModule() -> ModuleDeclSyntax? {
     do {
+      guard currentToken?.tokenKind == .moduleKeyword else {
+        throw Diagnostic.Message.expectedTopLevelModule
```

Comment:
We usually use `peek()` instead of `currentToken?.tokenKind`

Reply:
Makes sense 👍. I see it defaults to `0`. I was expecting it only returns future tokens

Reply:
Good catch on the double-diagnoses. 
I can register the error directly in the `else` block instead of throwing it?
```Swift
guard peek() == .moduleKeyword else {
  engine.diagnose(.expectedTopLevelModule, node: currentToken)
  return nil
}
```

Reply:
You can just `throw engine.diagnose(...)`. `diagnose` returns the message that was diagnosed for exactly this reason 😄

Reply:
Genius. Done. Sorry to be such high maintenance for a minor tweak 😅

Reply:
No, no, no don’t apologize! You’re doing great! Thanks so much for contributing.